DuggaSys/diagram.js

```diff
@@ -89,20 +89,35 @@ var pointerState = pointerStates.DEFAULT;
 var movingObject = false;
 var movingContainer = false;
 
+var randomidArray = []; // array for checking randomID
 //-------------------------------------------------------------------------------------------------
 // makeRandomID - Random hex number
 //-------------------------------------------------------------------------------------------------
 
 function makeRandomID()
 {
     var str = "";
+    var checkID = true;
     var characters = 'ABCDEF0123456789';
     var charactersLength = characters.length;
+    while(checkID == true){
```

Comment:
While statement could be just "true" since the while loop runs until the return statement, so CheckID == true kind of make no sense here.

Reply:
Indentation for the whole while statement/function needs to be overlooked.